### PR TITLE
[go] Added support for `1.22` and updated Go minor versions of `1.19`, `1.20` and `1.21`

### PR DESCRIPTION
**Pull request by StefMa**
Title ☝️ 🙃

### CODE REVIEW

**Comment by StefMa** [comment]
Seems only upload test reports are failing. But tests are good. So from my side it seems everything to work and ready to merge...

**Comment by StefMa** [comment]
@TooTallNate anything that prevents to merge this PR? 🤔 
Unfortunately I don't know how to fix the tests.. Maybe you can help out here?

**Comment by StefMa** [comment]
[Another Go PR](https://github.com/vercel/vercel/pull/11064) also stuck since month.

Go support at vercel is declared as beta since years too.

I have the feeling we should slowly search for alternative providers for Go serverless functions 👀

**Comment by vedantmgoyal9** [comment]
@StefMa please update the branch

**Comment by vedantmgoyal9** [comment]
@StefMa 1.22.2 was released, day 50 of asking vercel maintainers to merge... @TooTallNate @trek @jeffsee55 @onsclom @EndangeredMassa

**Comment by TooTallNate** [comment]
Sorry for the delay everyone. We'll be looking to get this merged soon.

In the meantime, you can set env var `VERCEL_CLI_VERSION` with value of `https://vercel-git-fork-stefma-go-support.vercel.sh/tarballs/vercel.tgz` to create a deployment using the changes in this PR.

File: packages/go/src/go-helpers.ts [hunk 1]
```diff
@@ -22,9 +22,10 @@ import type { Env } from '@vercel/build-utils';
 const streamPipeline = promisify(pipeline);
 
 const versionMap = new Map([
-  ['1.21', '1.21.1'],
-  ['1.20', '1.20.2'],
-  ['1.19', '1.19.6'],
+  ['1.22', '1.22.1'],
```
Comment (vedantmgoyal9) [inline packages/go/src/go-helpers.ts]:
```suggestion
  ['1.22', '1.22.2'],
```

File: packages/go/test/fixtures/01-cowsay/vercel.json [hunk 2]
```diff
@@ -5,10 +5,10 @@
     { "src": "subdirectory/index.go", "use": "@vercel/go" }
   ],
   "probes": [
-    { "path": "/", "mustContain": "cow:go1.21.1:RANDOMNESS_PLACEHOLDER" },
+    { "path": "/", "mustContain": "cow:go1.22.0:RANDOMNESS_PLACEHOLDER" },
```
Comment (vedantmgoyal9) [inline packages/go/test/fixtures/01-cowsay/vercel.json]:
```suggestion
    { "path": "/", "mustContain": "cow:go1.22.2:RANDOMNESS_PLACEHOLDER" },
```

File: packages/go/test/fixtures/01-cowsay/vercel.json [hunk 3]
```diff
@@ -5,10 +5,10 @@
...
     {
       "path": "/subdirectory",
-      "mustContain": "subcow:go1.21.1:RANDOMNESS_PLACEHOLDER"
+      "mustContain": "subcow:go1.22.0:RANDOMNESS_PLACEHOLDER"
```
Comment (vedantmgoyal9) [inline packages/go/test/fixtures/01-cowsay/vercel.json]:
```suggestion
      "mustContain": "subcow:go1.22.2:RANDOMNESS_PLACEHOLDER"
```

**Review by TooTallNate — APPROVED** [review]

**Comment by StefMa** [comment]
giphy

**Pull request merged**